core/jobs/types/base_validation_errors.py

```diff
@@ -49,11 +64,24 @@ def __init__(self, message, model_or_kind, model_id=None):
             raise ValueError('message must be a non-empty string')
 
         if model_id is None:
+            # Here, get_model_id and get_model_kind can only accept argument
+            # of type base_models.BaseModel but model_or_kind is defined as a
+            # type of Union[base_models.BaseModel, str]. So, to rule out the
+            # possibility of str of model_or_kind for mypy type checking.
+            # We asserted it as an Basemodel type.
+            assert isinstance(model_or_kind, base_models.BaseModel)
             model_id = job_utils.get_model_id(model_or_kind)
             model_kind = job_utils.get_model_kind(model_or_kind)
         else:
+            # The type of model_or_kind is Union[base_models.BaseModel, str]
+            # and the type of model_kind is str. So, to rule out the possibilty
+            # of Basemodel of model_or_kind for MyPy type checking. we asserted
+            # it as an str type.
+            assert isinstance(model_or_kind, str)
             model_kind = model_or_kind
 
+        # Ruling out the possibility of None for mypy type checking.
+        assert model_id is not None
```

Comment:
Looking at the code above it seems to me that `model_id` can be `None` in some cases, right?

Reply:
Yes, you are right.
Done!